pyscitt/pyscitt/key_vault_sign_client.py

```diff
@@ -0,0 +1,101 @@
+
+from azure.keyvault.certificates import (
+    CertificateClient,
+    KeyVaultCertificate
+)
+from azure.keyvault.keys import (
+    KeyClient
+    )
+from azure.keyvault.keys.crypto import (
+    CryptographyClient,
+    SignatureAlgorithm,
+)
+
+class _DERSignature(Sequence):
+    """Internal helper class for decoding AKV signature for CCF."""
+    componentType = NamedTypes(
+        NamedType("r", Integer()),
+        NamedType("s", Integer()),
+    )
+
+class KeyVaultSignClient():
+    """MemberIdentity implementation that uses Azure Key Vault."""
+
+    def __init__(self, akv_sign_configuration: str):
+        akv_sign_configuration_dict = json.loads(akv_sign_configuration)
+        self._vault_name = akv_sign_configuration_dict['keyVaultName']
+        self._vault_url = f"https://{self._vault_name}.vault.azure.net"
+        self._identity_certificate_name = akv_sign_configuration_dict['cetificateName']
+        self._identity_certificate_version = akv_sign_configuration_dict['cetificateVersion']
+        self._identity_private_curve_key_size = akv_sign_configuration_dict['privateCurveKeySize']
+    
+    @staticmethod
+    def _decode_certificate(cert: KeyVaultCertificate) -> str:
+        decoded = base64.b64encode(cert.cer).decode()  # type: ignore
+
+        styled_cert = "-----BEGIN CERTIFICATE-----\n"
+        for index in range(0, len(decoded), 64):
+            if index + 64 > len(decoded):
+                styled_cert += decoded[index:]
+            else:
+                styled_cert += decoded[index : index + 64]
+            styled_cert += "\n"
+        styled_cert += "-----END CERTIFICATE-----"
+        return styled_cert
+
+    def sign_with_idetity(self, digest: bytes):
+        credential = DefaultAzureCredential()
+        key_client = KeyClient(vault_url=self._vault_url, credential=credential)
+        cert_client = CertificateClient(vault_url=self._vault_url, credential=credential)
+
+        key = key_client.get_key(
+                        name=self._identity_certificate_name,
+                        version=self._identity_certificate_version,
+                    )
+                    
+        cert = cert_client.get_certificate_version(
+            certificate_name=self._identity_certificate_name,
+            version=self._identity_certificate_version
+            )
+
+        crypto_client = CryptographyClient(key, credential=credential)
+
+        # The signatures returned by AKV are returned as a JWS signature and encoded in
+        # base64url format and are not directly compatible with the signatures supported by CCF.
+        sign_result = None
+        signature_algorithm = SignatureAlgorithm.es384
+        # See https://github.com/microsoft/CCF/blob/master/doc/members/jws_to_der.py
+        # for original conversion code.
+        # digest_to_sign = hashlib.sha384(digest.encode()).digest()
+
+        digest_to_sign = {
+            256: hashlib.sha256(digest).digest(), 
```

Comment:
We should just use the default alg that matches the curve of the key. See also `default_algorithm_for_key` in the crypto module which takes a public or private key as input. In this case, it would be the public key from the cert.